Networking/WebSockets/WebSocketImpl.cc

```diff
@@ -97,13 +97,33 @@ namespace litecore { namespace websocket {
         startResponseTimer(chrono::seconds(kConnectTimeoutSecs));
     }
 
+    void WebSocketImpl::gotTLSCertificate(slice certData) {
+        std::lock_guard<std::mutex> lock(_mutex);
+
+        if (_closed) {
+            warn("WebSocket already closed, ignoring onWebSocketGotTLSCertificate...");
+            return;
+        }
+
+        logInfo("Got TLS certificate");
+        delegate().onWebSocketGotTLSCertificate(certData);
```

Comment:
You can't call the delegate while holding the mutex lock. It'll cause a deadlock if the delegate calls back into the WebSocket object. (Note other existing methods where we put the `lock_guard` in a nested block so it's gone by the time we call the delegate.)